components/cqrs/event_bus.go

```diff
@@ -2,18 +2,76 @@ package cqrs
 
 import (
 	"context"
+	stdErrors "errors"
 
+	"github.com/ThreeDotsLabs/watermill"
 	"github.com/ThreeDotsLabs/watermill/message"
 	"github.com/pkg/errors"
 )
 
+type EventBusConfig struct {
+	GeneratePublishTopic GenerateEventPublishTopicFn
+
+	// OnPublish is called before sending the event.
+	// The *message.Message can be modified.
+	//
+	// This option is not required.
+	OnPublish OnEventSendFn
+
+	// Marshaler is used to marshal and unmarshal events.
+	// It is required.
+	Marshaler CommandEventMarshaler
+
+	// Logger instance used to log.
+	// If not provided, watermill.NopLogger is used.
+	Logger watermill.LoggerAdapter
+}
+
+func (c *EventBusConfig) setDefaults() {
+	if c.Logger == nil {
+		c.Logger = watermill.NopLogger{}
+	}
+}
+
+func (c EventBusConfig) Validate() error {
+	var err error
+
+	if c.Marshaler == nil {
+		err = stdErrors.Join(err, errors.New("missing Marshaler"))
+	}
+
+	if c.GeneratePublishTopic == nil {
+		err = stdErrors.Join(err, errors.New("missing GenerateHandlerTopic"))
+	}
+
+	return err
+}
+
+type GenerateEventPublishTopicFn func(GenerateEventPublishTopicParams) (string, error)
```

Comment:
What's the use case for returning an error here?

Reply:
up to the person who implements that - I wanted to give an option to run some input validation if someone wants